### PR TITLE
[ios] apply safe area insets to Kodi GUI

### PR DESCRIPTION
**Pull request by kambala-decapitator**
## Description
Extends `RESOLUTION_INFO` with `guiInsets` field that adds edge insets when computing GUI scaling in `CGraphicContext::GetGUIScaling()`.

## Motivation and Context
This is a better approach to making GUI not being obscured by device's notch originally introduced in #16719, because it still allows rendering video in full screen. Originally requested at https://forum.kodi.tv/showthread.php?tid=348671.

## How Has This Been Tested?
Tested on iPhone XR (with notch) and iPhone 6+ / iPad Air 2 (without) and with external screen connected via HDMI adapter.

## Screenshots:
1_framed
1force_landscapeleft_framed
2_framed
2force_landscapeleft_framed

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by kambala-decapitator** [comment]
jenkins build this please

**Comment by kambala-decapitator** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]
Cannot runtime-test. Code looks good to me.

File: xbmc/windowing/Resolution.cpp [hunk 1]
```diff
@@ -20,6 +20,14 @@
 
 #include <cstdlib>
 
+EdgeInsets::EdgeInsets() : EdgeInsets(0.0f, 0.0f, 0.0f, 0.0f)
```
Comment (Rechi) [inline xbmc/windowing/Resolution.cpp]: Any reason for not using default member initialization?

Comment (kambala-decapitator) [reply xbmc/windowing/Resolution.cpp]: You mean using `= default`? Yes, good point! Thanks, will change.

Comment (Rechi) [reply xbmc/windowing/Resolution.cpp]: `float left = 0.0f;` in the struct definition is default member initialization. Once that is used you can then also use `= default`.

File: xbmc/windowing/Resolution.cpp [hunk 2]
```diff
@@ -20,6 +20,12 @@
 
 #include <cstdlib>
 
+EdgeInsets::EdgeInsets() = default;
```
Comment (Rechi) [inline xbmc/windowing/Resolution.cpp]: could be done in the header file

Comment (kambala-decapitator) [reply xbmc/windowing/Resolution.cpp]: I had an impression that adding definitions to the header is bad practice. Or trivial ones are ok? Will change.

Also a question: is `float left{0.0f};` or even `float left{};` better than `float left = 0.0f;`?

Comment (kambala-decapitator) [reply xbmc/windowing/Resolution.cpp]: @Rechi please check my question above

Comment (Rechi) [reply xbmc/windowing/Resolution.cpp]: IMO adding `= default` for constructors is okay in the header file.
Whatever you prefer.

Comment (kambala-decapitator) [reply xbmc/windowing/Resolution.cpp]: thanks!

**Comment by kambala-decapitator** [comment]
Jenkins build this please

**Comment by phunkyfish** [comment]
@kambala-decapitator is this ready to go in?

**Comment by kambala-decapitator** [comment]
@phunkyfish yes, just waiting for #17465 to be merged to have all lights green

**Comment by kambala-decapitator** [comment]
Jenkins build this please

**Pull request merged**